File: monitoring-service/src/db/models/activity_flow_period.rs
use time::OffsetDateTime;

#[derive(Debug, sqlx::FromRow)]
pub struct ActivityFlowPeriod {
    pub id: Option<i64>,
    pub start_time: Option<OffsetDateTime>,
    pub end_time: Option<OffsetDateTime>,
    pub score: f64,
    pub app_switches: i64,
    pub inactive_time: i64,
    pub created_at: OffsetDateTime,
}

File: src/db/activityFlowPeriod.ts
import { MonitorDb } from './monitorDb'

export interface ActivityFlowPeriod {
  id: number
  start_time: string
  end_time: string
  score: number
  app_switches: number
  inactive_time: number
  created_at: string
}

const getActivityFlowPeriodsBetween = async (start: string, end: string) => {
  const monitorDb = await MonitorDb.getMonitorDb()
  const activityFlowPeriods = await monitorDb.select<ActivityFlowPeriod[]>(
    `SELECT * FROM activity_flow_period WHERE start_time >= '${start}' AND end_time <= '${end}';`,
  )
  return activityFlowPeriods
}

export const ActivityFlowPeriodDb = {
  getActivityFlowPeriodsBetween,
}
